components/centraldashboard/app/api_workgroup.ts

```diff
@@ -244,6 +234,16 @@ export class WorkgroupApi {
                 });
             }
         })
+        .use((req: Request, res: Response, next: NextFunction) => {
+            if (!req.user.hasAuth) {
+                return apiError({
+                    res,
+                    code: 405,
+                    error: 'Access to this api is denied',
```

Comment:
You may want to make the error message more descriptive if you intend on surfacing it to the user. Perhaps something that suggests that the user's identity could not be determined from the request.

Reply:
Makes sense, done!